### PR TITLE
feature/remove sysattrs from attributes

### PR DESCRIPTION
**Pull request by gpoujol**

### CODE REVIEW

File: search-service/src/main/kotlin/db/migration/V0_28__JsonLd_migration.kt [hunk 1]
```diff
@@ -90,10 +91,11 @@ class V0_28__JsonLd_migration : BaseJavaMigration() {
 
             val jsonLdEntity = JsonLdEntity(expandedEntity, contexts)
             val ngsiLdEntity = jsonLdEntity.toNgsiLdEntity()
+
             // in current implementation, geoproperties do not have a creation date as they are stored
             // as a property of the entity node, so we give them the creation date of the entity
             val defaultCreatedAt = getPropertyValueFromMapAsDateTime(
-                expandedEntity as Map<String, List<Any>>,
+                originalExpandedEntity as Map<String, List<Any>>,
```
Comment (bobeal) [inline search-service/src/main/kotlin/db/migration/V0_28__JsonLd_migration.kt:98]: what does it change?

File: search-service/src/main/kotlin/db/migration/V0_28__JsonLd_migration.kt [hunk 2]
```diff
@@ -107,13 +109,20 @@ class V0_28__JsonLd_migration : BaseJavaMigration() {
                         datasetId
                     )!! as Map<String, List<Any>>
 
+                    val attributePayloadFiltered = attributePayload
+                        .filterKeys { attributeName ->
+                            // remove createdAt and modifiedAt in attribute's payload
+                            attributeName != NGSILD_CREATED_AT_PROPERTY
+                            attributeName != NGSILD_MODIFIED_AT_PROPERTY
```
Comment (bobeal) [inline search-service/src/main/kotlin/db/migration/V0_28__JsonLd_migration.kt:116]: only the second line will be taken into account

File: shared/src/main/kotlin/com/egm/stellio/shared/util/JsonLdUtils.kt [hunk 3]
```diff
@@ -113,9 +113,8 @@ object JsonLdUtils {
                 expandContext = mapOf(JSONLD_CONTEXT to addCoreContext(contexts))
             }
 
-    fun expandDeserializedPayload(deserializedPayload: Map<String, Any>, contexts: List<String>): Map<String, Any> {
-        return doJsonLdExpansion(deserializedPayload, contexts)
-    }
+    fun expandDeserializedPayload(deserializedPayload: Map<String, Any>, contexts: List<String>): Map<String, Any> =
+        doJsonLdExpansion(deserializedPayload, contexts)
```
Comment (bobeal) [inline shared/src/main/kotlin/com/egm/stellio/shared/util/JsonLdUtils.kt:117]: I know it's more elegant as a one-liner but this should not be part of the PR

File: search-service/src/main/kotlin/db/migration/V0_28__JsonLd_migration.kt (hunk 1 above)
Comment (gpoujol) [reply search-service/src/main/kotlin/db/migration/V0_28__JsonLd_migration.kt:98]: I remove this

File: shared/src/main/kotlin/com/egm/stellio/shared/util/JsonLdUtils.kt (hunk 3 above)
Comment (gpoujol) [reply shared/src/main/kotlin/com/egm/stellio/shared/util/JsonLdUtils.kt:117]: I do it to correct the error

**Review by bobeal — APPROVED** [review]

**Pull request merged**